static/js/settings_users.js

```diff
@@ -482,8 +483,13 @@ function handle_deactivation($tbody) {
         const user_id = $row.data("user-id");
 
         function handle_confirm() {
+            const is_spammer = JSON.stringify($("#mark_as_spammer").prop("checked"));
+            const delete_action = $("#message_delete_action")[0].value;
             const url = "/json/users/" + encodeURIComponent(user_id);
-            dialog_widget.submit_api_request(channel.del, url);
+            dialog_widget.submit_api_request(channel.del, url, {
+                spammer: is_spammer,
+                message_delete_action: delete_action,
+            });
```

Comment:
Reading this I noticed this bug: 

Fixed in 43a57494f90c98022e20913c37f81c4d186e4614, including rebasing your commit on top of that.